editor/src/document/document_file.rs

```diff
@@ -372,12 +372,29 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 				Err(e) => log::error!("DocumentError: {:?}", e),
 				Ok(_) => (),
 			},
-			RenderDocument => responses.push_back(
+			RenderDocument => responses.extend([
 				FrontendMessage::UpdateCanvas {
 					document: self.document.render_root(),
 				}
 				.into(),
-			),
+				FrontendMessage::UpdateScrollbars {
+					bounds: {
+						let bounds = self.active_document_mut().document.visible_layers_bounding_box();
+						let bounds = bounds.unwrap_or([glam::DVec2::ZERO, glam::DVec2::ZERO]);
+						let viewport = ipp.viewport_size.as_f64();
+						[
+							bounds[0].x.min(0.) - viewport.x * 0.5,
+							bounds[0].y.min(0.) - viewport.y * 0.5,
+							bounds[1].x.max(viewport.x) + viewport.x * 0.5,
+							bounds[1].y.max(viewport.y) + viewport.y * 0.5,
+						]
+					},
```

Comment:
This is not final yet, but in general I would prefer something like this:
```rust
                        let bounds = bounds.unwrap_or([glam::DVec2::ZERO, glam::DVec2::ZERO]);
						let [pos1, pos2] = ipp.viewport_size.as_f64();
                        pos1 = pos1.min(DVec::Zero) - viewport * 0.5;
                        pos2 = pos2.max(viewport) + viewport * 0.5;
						[ pos1.x, pos1.y, pos2.x, pos2.y]

Reply:
while you're at it you could also remove the .active_document() functions to avoid future confusion. Was a bit lazy and did not fix everything the the transforms pr

Reply:
Done